dashboard/app/helpers/levels_helper.rb

```diff
@@ -178,7 +179,7 @@ def app_options
 
     if (@level.channel_backed? && params[:action] != 'edit_blocks') || @level.is_a?(Javalab)
       view_options(
-        channel: get_channel_for(@level, @user),
+        channel: get_channel_for(@level, @script&.id, @user),
```

Comment:
from the Jira epic, it looks like you are planning to add a not-null constraint on script_id in the future. This suggests to me that you'll need to come up with a plan for what to do on lines like this for when `@script` is nil. cases which come to mind include (1) standalone projects, and (2) when viewing a level by id in levelbuilder mode. I'm not sure we have tests covering the latter. I'll see if I can add one.

Reply:
https://github.com/code-dot-org/code-dot-org/pull/42167

Reply:
So the ticket with the non-null constraint was resolved with status "won't do" https://codedotorg.atlassian.net/browse/LP-1846, for reason number 2 that you mentioned which is that in levelbuilder we'll be able to view a level that's channel-backed outside of the context of a script so script_id will be nil.

Reply:
Great! No concerns with instances of `@script&.id`, then 👍